packages/docusaurus-plugin-content-docs/src/client/docsClientUtils.ts

```diff
@@ -59,17 +59,38 @@ export function getActivePlugin(
 export const getLatestVersion = (data: GlobalPluginData): GlobalVersion =>
   data.versions.find((version) => version.isLast)!;
 
+export function sortVersionsByPathDepth(
+  versions: GlobalVersion[],
+): GlobalVersion[] {
+  return [...versions].sort((a, b) => {
+    const getDepth = (path: string): number => {
+      if (path === '/') {
+        return 0;
+      }
+      const trimmedPath = path.replace(/^\/|\/$/g, '');
+      return trimmedPath ? trimmedPath.split('/').length : 0;
+    };
+
+    const depthA = getDepth(a.path);
+    const depthB = getDepth(b.path);
+
+    // Sort by depth (descending order)
+    if (depthA !== depthB) {
+      return depthB - depthA;
+    }
+
+    // else sort alphabetically
+    return a.path.localeCompare(b.path);
+  });
+}
```

Comment:
FYI we have a `sortRoutes` fn on Docusaurus core already

This part of the algo looks relevant for our case here, and simpler than what you did

![CleanShot 2024-07-22 at 12 49 13](https://github.com/user-attachments/assets/c5ddd40b-a7d2-4990-9aef-132179a8baaf)